nameable.rb

```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def self.correct_name
+    raise NotImplementedError
+  end
```

Comment:
- [ **OPTIONAL** ] **RECOMMENDED:** I think it would be better if you update your `correct_name` method by adding the full message to help the `User` to correctly understand the error. You can update to something like this below:
```ruby
class Nameable
    def correct_name
        raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
    end
end
```